cmd/adapter/query/query.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package query
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/alecthomas/participle"
+	"github.com/alecthomas/participle/lexer"
+	"strconv"
+	"strings"
+)
+
+type (
+	// Represents parsed 'Gravity log query'
+	query struct {
+		Exp *expression `@@`
```

Comment:
This does not conform to struct tag syntax and will issue a warning.
Consider prefixing this with `parser` (See https://github.com/alecthomas/participle/issues/32)

Reply:
applied